### PR TITLE
fix: do not allow additional properties in output object

### PR DESCRIPTION
**Pull request by ytausch**
This would fix the issue described in #12 but breaks YAMLs that should be valid (tests are failing). Intended as a basis for discussion.

### CODE REVIEW

File: examples/invalid/complex1.yaml
Comment (pavelzw) [inline examples/invalid/complex1.yaml:1]: Let's add a readme to this directory stating that these files are only for testing and _not_ valid examples.

File: examples/invalid/complex1.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+      name: my-package
+      version: '0.0.1'
+    build:
+      noarch: python
+      script:
+        - echo hello world
+    testaaaa: # this should fail but doesn't
+      - python:
+          imports:
+            - polarify
```
Comment (pavelzw) [inline examples/invalid/complex1.yaml:13]:
```suggestion
            - polarify

```

**Review by pavelzw — APPROVED** [review]

**Comment by pavelzw** [comment]
> breaks YAMLs that should be valid (tests are failing)

Actually these yamls are not valid too 😅 

[fixes.patch](https://github.com/prefix-dev/recipe-format/files/14303400/fixes.patch)
if you apply this patch, the examples should be in sync with the latest recipe format :)

**Comment by pavelzw** [comment]
@wolfv PTAL

**Review by baszalmstra — APPROVED** [review]
Nice! Im glad you were able to fix the examples too!

**Comment by wolfv** [comment]
Great stuff! :)

**Pull request merged**